[Profiler][Minor] Extend Python bindings

robieta opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #83965
* #83964
* #83895
* #83894
* #83893
* #83892
* #83891
* #83890
* #83963
* __->__ #83622

Adding some fields which are needed for memory profiling.

Differential Revision: [D38528382](https://our.internmc.facebook.com/intern/diff/D38528382/)

Commit:
[Profiler][Minor] Extend Python bindings

Adding some fields which are needed for memory profiling.

Differential Revision: [D38528382](https://our.internmc.facebook.com/intern/diff/D38528382/)

[ghstack-poisoned]

Commit:
Update on "[Profiler][Minor] Extend Python bindings"


Adding some fields which are needed for memory profiling.

Differential Revision: [D38528382](https://our.internmc.facebook.com/intern/diff/D38528382/)

[ghstack-poisoned]

Gamrix commented on torch/csrc/profiler/python/init.cpp:
```diff
@@ -122,19 +135,32 @@ void initPythonBindings(PyObject* module) {
+  py::class_<torch_op_t>(m, "_ExtraFields_TorchOp")
+      .def_readonly("inputs", &torch_op_t::inputs_)
+      .def_readonly("scope", &torch_op_t::scope_)
+      .def_readonly("sequence_number", &torch_op_t::sequence_number_)
```
When looking in the codebase, I didn't find a docstring documenting what this `sequence_number` is supposed to hold. Could you add a docstring to where the C++ field is declared to explain the fields and their properties in a separate diff?

Gamrix commented on torch/csrc/profiler/python/init.cpp:
```diff
@@ -122,19 +135,32 @@ void initPythonBindings(PyObject* module) {
+            return reinterpret_cast<intptr_t>(a.ptr_);
+          })
+      .def_readonly("alloc_size", &allocation_t::alloc_size_)
+      .def_readonly("device_type", &allocation_t::device_type_)
+      .def_readonly("device_index", &allocation_t::device_index_)
```
Do we want to reassemble a `device_type` and a `device_index` back into a `Device` so that this API is more accessible to end users?

Gamrix reviewed (approved):
Looks good, I left a few small comments on things to consider.

Commit:
Update on "[Profiler][Minor] Extend Python bindings"


Adding some fields which are needed for memory profiling.

Differential Revision: [D38528382](https://our.internmc.facebook.com/intern/diff/D38528382/)

[ghstack-poisoned]

Commit:
Update on "[Profiler][Minor] Extend Python bindings"


Adding some fields which are needed for memory profiling.

Differential Revision: [D38528382](https://our.internmc.facebook.com/intern/diff/D38528382/)

[ghstack-poisoned]

Commit:
Update on "[Profiler][Minor] Extend Python bindings"


Adding some fields which are needed for memory profiling.

Differential Revision: [D38528382](https://our.internmc.facebook.com/intern/diff/D38528382/)

[ghstack-poisoned]

Commit:
Update on "[Profiler][Minor] Extend Python bindings"


Adding some fields which are needed for memory profiling.

Differential Revision: [D38528382](https://our.internmc.facebook.com/intern/diff/D38528382/)

[ghstack-poisoned]

Commit:
Update on "[Profiler][Minor] Extend Python bindings"


Adding some fields which are needed for memory profiling.

Differential Revision: [D38528382](https://our.internmc.facebook.com/intern/diff/D38528382/)

[ghstack-poisoned]

Commit:
Update on "[Profiler][Minor] Extend Python bindings"


Adding some fields which are needed for memory profiling.

Differential Revision: [D38528382](https://our.internmc.facebook.com/intern/diff/D38528382/)

[ghstack-poisoned]

Commit:
Update on "[Profiler][Minor] Extend Python bindings"


Adding some fields which are needed for memory profiling.

Differential Revision: [D38528382](https://our.internmc.facebook.com/intern/diff/D38528382/)

[ghstack-poisoned]

Commit:
Update on "[Profiler][Minor] Extend Python bindings"


Adding some fields which are needed for memory profiling.

Differential Revision: [D38528382](https://our.internmc.facebook.com/intern/diff/D38528382/)

[ghstack-poisoned]

Commit:
Update on "[Profiler][Minor] Extend Python bindings"


Adding some fields which are needed for memory profiling.

Differential Revision: [D38528382](https://our.internmc.facebook.com/intern/diff/D38528382/)

[ghstack-poisoned]

Commit:
Update on "[Profiler][Minor] Extend Python bindings"


Adding some fields which are needed for memory profiling.

Differential Revision: [D38528382](https://our.internmc.facebook.com/intern/diff/D38528382/)

[ghstack-poisoned]

Commit:
Update on "[Profiler][Minor] Extend Python bindings"


Adding some fields which are needed for memory profiling.

Differential Revision: [D38528382](https://our.internmc.facebook.com/intern/diff/D38528382/)

[ghstack-poisoned]

Commit:
Update on "[Profiler][Minor] Extend Python bindings"


Adding some fields which are needed for memory profiling.

Differential Revision: [D38528382](https://our.internmc.facebook.com/intern/diff/D38528382/)

[ghstack-poisoned]
